addon/editor/input-handlers/text-input-handler.ts

```diff
@@ -91,171 +67,43 @@ export default class TextInputHandler implements InputHandler {
-        this.rawEditor.setCaret(textNode, textNode.length);
-      }
-    }
-    else if (manipulation.type == "replaceSelectionWithText") {
-      const { selection, text } = manipulation;
-      const range = selection.getRangeAt(0);
-      if (range) {
-        const {startContainer, endContainer, startOffset, endOffset } = range;
-        if (startContainer.nodeType == Node.TEXT_NODE) {
-          selection.deleteFromDocument();
-          insertTextIntoTextNode(startContainer as Text, startOffset, text);
-          this.rawEditor.updateRichNode();
-          this.rawEditor.setCaret(startContainer, startOffset + text.length);
-        }
-        else if (endContainer.nodeType == Node.TEXT_NODE) {
-          selection.deleteFromDocument();
-          insertTextIntoTextNode(endContainer as Text, endOffset, text);
-          this.rawEditor.updateRichNode();
-          this.rawEditor.setCaret(endContainer, endOffset + text.length);
-        }
-        else {
-          const textNode = document.createTextNode(text);
-          selection.deleteFromDocument();
-          startContainer.childNodes[startOffset - 1].after(textNode);
-          this.rawEditor.updateRichNode();
-          this.rawEditor.setCaret(textNode, textNode.length);
-        }
-      }
-      else {
-        console.warn("no selected range, not doing anything!");
-      }
-    }
-
-    else {
-      throw "unsupport manipulation";
+  handleNativeManipulation(manipulation: TextHandlerManipulation) {
+    console.log("handling native manip");
+    if (manipulation.type === "insertTextIntoRange") {
+      console.log(manipulation);
+      this.rawEditor.executeCommand("insert-text", manipulation.text, manipulation.range);
+    } else {
+      throw new UnsupportedManipulationError(manipulation);
     }
   }
 
-  getNextManipulation(event: KeyboardEvent) : Manipulation {
-    const selection = window.getSelection();
-    if (selection?.isCollapsed) {
-      const { anchorNode, anchorOffset } = selection;
-      if (anchorNode?.nodeType == Node.TEXT_NODE) {
-        return { type: "insertTextIntoTextNode", node: anchorNode as Text, position: anchorOffset, text: event.key };
-      }
-      else if (anchorNode?.nodeType == Node.ELEMENT_NODE) {
-        return { type: "insertTextIntoElement", node: anchorNode as HTMLElement, position: anchorOffset, text: event.key};
-      }
-      else {
-        throw "unsupported selection";
-      }
+  getNextManipulation(event: KeyboardEvent): TextHandlerManipulation {
+    const range = this.rawEditor.model.selection.lastRange;
+    if (!range) {
+      throw new MisbehavedSelectionError();
     }
-    else if (selection && this.rawEditor.rootNode.contains(selection.anchorNode)) {
-      return { type: "replaceSelectionWithText", selection: selection, node: (selection.anchorNode as Node), text: event.key };
-    }
-    throw "selection is required for text input";
+    const text = event.key === " " ? NON_BREAKING_SPACE : event.key;
```

Comment:
I assume this is an attempt at simplifying the logic wrt to spaces, but this will result in all spaces being non breaking spaces. Which means our lines will not break if you type a long sentence in the editor. Unless this is addressed in the writer or the mutator I don't think this is a good solution

Reply:
should be handled in https://github.com/lblod/ember-rdfa-editor/pull/102/commits/6a51855654eb2709c0c100a67254990a628db05f